src/libraries/Directory.Build.targets

```diff
@@ -241,14 +241,10 @@
     </ItemGroup>
   </Target>
 
-  <!-- Adds Nullable annotation attributes to netstandard <= 2.0 builds -->
-  <Choose>
-    <When Condition="'$(Nullable)' != '' and ($(TargetFramework.StartsWith('netstandard1')) or '$(TargetFramework)' == 'netstandard2.0' or $(TargetFramework.StartsWith('netcoreapp2')) or '$(TargetFrameworkIdentifier)' == '.NETFramework')">
-      <ItemGroup>
-        <Compile Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\NullableAttributes.cs" Link="System\Diagnostics\CodeAnalysis\NullableAttributes.cs" />
-      </ItemGroup>
-    </When>
-  </Choose>
+  <!-- Adds Nullable annotation attributes to < .NET 5 builds. -->
+  <ItemGroup Condition="'$(Nullable)' != '' and !$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net5.0'))">
+    <Compile Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\NullableAttributes.cs" Link="System\Diagnostics\CodeAnalysis\NullableAttributes.cs" />
+  </ItemGroup>
```

Comment:
Doesn't this mean anywhere we have a netcoreapp3.0 build we're going to add these attributes even though most of them already exist in .NET Core 3.0?

Reply:
You already noticed in another comment that we are including that file everywhere but conditionally excluding some of the attributes that already shipped. That's especially useful when upgrading the .NETCoreApp assets (in a subsequent PR).